src/common/config/metadata/type.go

```diff
@@ -0,0 +1,140 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/goharbor/harbor/src/common/utils/log"
+)
+
+// Type - Use this interface to define and encapsulate the behavior of validation and transformation
+type Type interface {
+	// validate the configure value
+	validate(str string) error
+	// get the real type of current value, if it is int, return int, if it is string return string etc.
+	get(str string) interface{}
```

Comment:
why doesn't return an error in get()? Define a default return value, like 0, false is not accepted.

The caller could get 0 for any unknown int key, this does not make sense.

Reply:
Avoid return error in get to make the api call more simple, the error will be logged in background, because all configure related error should not be handled by end user.